packages/rx_bloc_cli/mason_templates/bricks/lib_auth_matrix/__brick__/lib/lib_auth_matrix/services/auth_matrix_service.dart

```diff
@@ -1,160 +1,108 @@
-      }
-    }
-    return;
-  }
 
-  ///Function used to verify the auth matrix flow
-  ///Adds the response to the [_flowEvents] stream
-  Future<void> verifyAuthMethod(
-    AuthMatrixResponse authMatrixResponse,
-    String endToEndId,
-    String userData,
-    String code,
-  ) async {
-    _flowEvents.add(
-      await _authMatrixRepository
-          .verifyAuthMatrix(
-        payload: AuthMatrixVerify(
-          transactionId: authMatrixResponse.transactionId,
-          userData: userData,
-          payload: {'code': code},
-          action: authMatrixResponse.authZ,
-          endToEndId: endToEndId,
-        ),
-      )
-          .catchError(
-        (error, stacktrace) {
-          _flowEvents.addError(
-            AccessDeniedErrorModel(),
-          );
-          throw AccessDeniedErrorModel();
-        },
-      ),
+  final BehaviorSubject<AuthMatrixResponse> _onResponse = BehaviorSubject();
+
+  /// Initiates the auth matrix process by the given [request].
+  ///
+  /// - [request] is the request body that contains the necessary user data to initiate the process.
+  /// Returns a [Stream] of [AuthMatrixResponse] that emits each step of the auth matrix process.
+  Stream<AuthMatrixResponse> initiateAuthMatrix({
+    required AuthMatrixPayloadRequest payload,
+  }) async* {
+    final response = await _authMatrixRepository.initiate(
+      action: payload.type,
+      request: payload,
     );
-  }
 
-  ///Function used to cancel the auth matrix flow
-  ///Adds the error to the [_flowEvents] stream
-  Future<void> cancelAuthMatrix(
-    String endToEndId,
-    String transactionId,
-  ) async {
-    await _authMatrixRepository.cancelAuthMatrix(
-      AuthMatrixCancelModel(endToEndId, transactionId),
-    );
-    _flowEvents.addError(AuthMatrixCanceledErrorModel());
+    yield* _executeAuthMethods(response);
   }
 
-  void dispose() {
-    _flowEvents.close();
-  }
+  /// Executes the auth matrix process by the given [response].
+  /// - [response] is the response that contains the necessary data to execute the auth matrix process.
+  /// Returns a [Stream] of [AuthMatrixResponse] that emits each step of the auth matrix process.
+  Stream<AuthMatrixResponse> _executeAuthMethods(
+    AuthMatrixResponse response,
+  ) async* {
+    AuthMatrixResponse? lastResponse = response;
+
+    while (true) {
+      if (lastResponse == null) {
+        // Complete the stream if the auth method returns null
+        break;
+      }
 
-  @override
-  ///Function used to verify auth method once the user enters the OTP code
-  Future<dynamic> confirmPhoneCode(String code) async {
-    await verifyAuthMethod(_flowEvents.value, _endToEndId, _userData, code);
-    return true;
-  }
+      yield lastResponse;
 
-  @override
-  Future<String> getFullPhoneNumber() async => '+1234567890';
+      // Emit the last response to the stream when the auth method is completed.
+      // this is exposed through the [onAuthenticationStepComplete] stream.
```

Comment:
I can't find an `onAuthenticationStepComplete`, I guess this was renamed to `onResponse`?